lighthouse-core/report/html/renderer/util.js

```diff
@@ -280,6 +280,79 @@ class Util {
     return parts.join(' ');
   }
 
+  /**
+   * Split a string by markdown code spans (enclosed in `backticks`), splitting
+   * into segments that were enclosed in backticks (marked as `isCode === true`)
+   * and those that outside the backticks (`isCode === false`).
+   * @param {string} text
+   * @return {Array<{isCode: true, codeText: string}|{isCode: false, plainText: string}>}
```

Comment:
We had a massive discussion about this very thing when discussing the shape of the proto and the hardened LHR API.

I 100% agree this feels anti-JS :)

OTOH, it's for internal use only, never goes over any wire, and we never really need to use just the text, so I don't have a strong reason to object yet.

Reply:
haha everyone lives with "preambleText" (does code typically come with a preamble?) including preambleText that occurs *after* the content and may not have any text in it, but I try to make self describing property names and everyone loses their minds :P

I think there is value in API safety like this (in this case tsc requires the `isCode` discriminator to be checked for the consuming code to be able to know if it can use `plainText` or `codeText`) but as you say it's internal only, for a minor feature, and I was really just playing around with the interface (not sure I really like it), so I can switch back to `text` or whatever :)